### ISSUE TITLE
Member Module- Additional Functionalities

### ISSUE THREAD

**Issue by Jeff254** [Dolibarr/dolibarr#28873]
### Feature Request

1. Fine/ Penalty for late Contribution.
2. Ability to define other Contributions.
3. Member Reports and Alerts.
4. Loans to Members: Members can receive and repay loans.

### Use case

Hello all,

Although the Member Module is a good feature, it lacks key functionalities to make it a full solution for Member Management especially for use by Groups/ Associations

The following are key functionalities necessary for the Module to be used by groups/ associations:

1. Member Contributions: This is currently there but it lacks for recurring monthly subscriptions and where there is a fine for late contribution.
2. Contributions: Ability to define other Contributions
3. Member Management: Should be easy to track which members have not paid which dues. In addition, it should also alert Members when a contribution has yet to be remitted or they have been fined
4. Loans to Members: Members can receive and repay loans.

**Comment by JonBendtsen** [comment]
@Jeff254 

> Contributions: Ability to define other Contributions

What is _other Contributions_?

**Comment by Jeff254** [comment]
In an Association you can have other Contributions for instance
1. Investment Contribution
2. Welfare Contribution

**Comment by JonBendtsen** [comment]
> In an Association you can have other Contributions for instance
> 1. Investment Contribution
> 2. Welfare Contribution


How would those contributions be valuated? Wouldn't they just be recalculated to a monetary contribution?

**Comment by Jeff254** [comment]
I don't understand your comment.

By Contribution I mean Monetary Contributions for instance;
1. Investment Contribution: $10
2. Welfare Contribution: $5
3. If a person remits their Investment Contribution after the set date then they are liable to $2 fine for late contribution. There is no fine for Welfare Contribution.
4. The Investment Contribution is remitted to Bank A while Welfare Contribution goes to Bank B.
5. Should be able to generate a report of the Contribution Status..

**Comment by JonBendtsen** [comment]
> I don't understand your comment.
> 
> By Contribution I mean Monetary Contributions for instance;

okay, I figured that investment contribution was some shares or bonds, and welfare would be to do some volunteer work.

Anyway, isn't this just different types of memberships? Or categories of membership. How is what you are asking for different?

When you record a contribution in Dolibarr you can so far only specify 1 product/service. Is it that you wish was multiple? Maybe with different membership types choosing individual products/services?

**Comment by Jeff254** [comment]
Yes, these are different categories of Membership where a Member can be in more than one (see No. 2). That's why I am requesting for 
1. Fine/ Penalty for late Contribution. 
2. Ability to define Multiple Contributions.
3. Member Reports and Alerts.
4. Loans to Members: Members can receive and repay loans.

**Comment by JonBendtsen** [comment]
I think that you are asking for a very specific usecase that is basically just your usecase.

> Yes, these are different categories of Membership where a Member can be in more than one (see No. 2). That's why I am requesting for
> 
> 1. Fine/ Penalty for late Contribution.

Since contributions can just be an invoice, just add a fee/another invoice if they don't pay in time. Sure it would be nice to have it automated - and you could probably already do this using the API or possibly a n8n integration.

https://wiki.dolibarr.org/index.php?title=Integration_of_Dolibarr_to_the_n8n_automation_platform

> 2. Ability to define multiple Contributions.

I think you can already do that.

![2_membership_contributions](https://github.com/Dolibarr/dolibarr/assets/2551481/bb2a21ca-e343-4887-bd4d-541e7cf036c0)

but okay, it just extends beyond the current contribution

![third_membership_contribution](https://github.com/Dolibarr/dolibarr/assets/2551481/81e9a23f-eddd-4022-bebe-c5ed5a348c6e)


> 3. Member Reports and Alerts.

Reports? What is that emails?

I can get a list of several different membership statuses: Draft, Validated, Pending, Up-to-date, Out-of-date, terminated, excluded.

![up2date_members](https://github.com/Dolibarr/dolibarr/assets/2551481/d297dc56-757d-4aac-86c3-5178fdc5e4c0)

I can get statistics

![member_statistics](https://github.com/Dolibarr/dolibarr/assets/2551481/14ddccfb-65de-4601-a234-ddf8c52c32de)

I can get a list of contributions

![member_contribution_list](https://github.com/Dolibarr/dolibarr/assets/2551481/335e50d1-b1d1-4711-9164-3712e3ff0bbe)

> 4. Loans to Members: Members can receive and repay loans.

I think Dolibarr already have a loan module - can't you use that?
![loan_module](https://github.com/Dolibarr/dolibarr/assets/2551481/2a2e12d2-89d5-4081-9673-bfb093934679)

**Comment by Jeff254** [comment]
### I think that you are asking for a very specific usecase that is basically just your usecase.

Is it now... I believe these are things that can enrich the membership module, especially where there are recurring contributions . Or maybe I don't understand it's objective then..

### I think you can already do that.
I don't think you can define and apply multiple contributions.

### Reports? What is that emails?
Reports just within Dolibarr.. basically fine tuning what you have shown. I want to be able to generate a report of Members who are yet to contribute.

### I think Dolibarr already have a loan module - can't you use that?
In my understanding the Loan Module is meant to be external loans to or from the company.
What I want is Management of loans to Members.
